application/src/main/java/com/connexta/replication/application/ConfigFileReader.java

```diff
@@ -0,0 +1,180 @@
+          LOGGER.info("New site \"{}\" saved", name);
+        }
+        // remove a site from the list if it exists in sites.json
+        existingSites.removeIf(s -> s.getName().equals(name));
+      }
+      // any remaining sites don't exist in sites.json so delete them
+      for (ReplicationSite site : existingSites) {
+        siteManager.remove(site.getId());
+        LOGGER.info("Site \"{}\" deleted", site.getName());
+      }
+
+      LOGGER.trace("Successfully read sites");
+    } catch (FileNotFoundException e) {
+      LOGGER.warn("Sites file not found");
+    } catch (JsonSyntaxException jse) {
+      LOGGER.warn("Sites file contained syntax errors", jse);
+    }
+  }
+
+  @VisibleForTesting
+  void readConfigs(Path configsPath) {
+    LOGGER.trace("Attempting to read configs from: {}", configsPath.toString());
+    try {
+      JsonReader jsonReader = new JsonReader(new FileReader(configsPath.toString()));
+      Gson gson = new Gson();
+      ReplicatorConfigImpl[] configs = gson.fromJson(jsonReader, ReplicatorConfigImpl[].class);
+      List<ReplicatorConfig> existingConfigs = configManager.objects().collect(Collectors.toList());
+      List<ReplicationSite> existingSites = siteManager.objects().collect(Collectors.toList());
+      for (ReplicatorConfig config : configs) {
+        String name = config.getName();
+        if (existingConfigs.stream().map(ReplicatorConfig::getName).noneMatch(name::equals)) {
+          // easier to specify names so the names of the sites will be in source and destination at
+          // first
+          // we need to swap them out for the ids
+          ReplicationSite source =
+              existingSites
+                  .stream()
+                  .filter(site -> site.getName().equals(config.getSource()))
+                  .findFirst()
+                  .orElse(null);
+          ReplicationSite destination =
+              existingSites
+                  .stream()
+                  .filter(site -> site.getName().equals(config.getDestination()))
+                  .findFirst()
+                  .orElse(null);
+
+          if (source == null) {
+            LOGGER.warn("site {} does not exist", config.getSource());
+            continue;
+          } else if (destination == null) {
+            LOGGER.warn("site {} does not exist", config.getDestination());
+            continue;
+          }
+          config.setSource(source.getId());
+          config.setDestination(destination.getId());
+          config.setFailureRetryCount(5);
+
+          configManager.save(config);
+          LOGGER.info("New config \"{}\" saved", name);
+        }
+        // remove a config from the list if it exists in configs.json
+        existingConfigs.removeIf(c -> c.getName().equals(name));
+      }
+      // any remaining configs don't exist in configs.json so delete them
+      for (ReplicatorConfig config : existingConfigs) {
+        configManager.remove(config.getId());
```

Comment:
Don't need to do anything right now but wonder if we should do any additional clean up when a config is removed?